### PR TITLE
#CU-861yuv dev: Create login page

### PR DESCRIPTION
**Pull request by thienduc12311**
Screen Shot 2021-06-12 at 18 14 05

### CODE REVIEW

**Comment by binh1298** [comment]
Task linked: [CU-861yuv Login Page](https://app.clickup.com/t/861yuv)

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -1,51 +1,51 @@
...
+            - WEB_ADMIN_ROOT_URL
+            - HASURA_ROOT_URL
+            - HASURA_GRAPHQL_DATABASE_URL
+            - GOOGLE_CLIENT_ID
+            - GOOGLE_SECRET
+            - GOOGLE_AUTH_URI
+            - JWT_SECRET
 volumes:
-  db_data:
+    db_data:
```
Comment (binh1298) [inline docker/docker-compose.yml:51]: Please remove this spacing

File: src/components/Buttons/ButtonBase/ButtonBase.stories.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { Story } from "@storybook/react";
 
 //👇 This default export determines where your story goes in the story list
 export default {
-    title: "Components/ButtonBase",
+    title: "Components/Button/ButtonBase",
```
Comment (binh1298) [inline src/components/Buttons/ButtonBase/ButtonBase.stories.tsx:9]:
```suggestion
    title: "Components/Buttons/ButtonBase",

```

File: src/components/Buttons/FabBase/FabBase.stories.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import FabBase, { FabBaseProps } from ".";
 import { Story } from "@storybook/react";
 
 export default {
-    title: "Components/FabBase",
+    title: "Components/Button/FabBase",
```
Comment (binh1298) [inline src/components/Buttons/FabBase/FabBase.stories.tsx:8]:
```suggestion
    title: "Components/Buttons/FabBase",
```

File: src/pages/Login/index.tsx [hunk 4]
```diff
@@ -73,14 +96,107 @@ const Login = () => {
...
     return (
-        <BoxBase>
-            <ButtonBase variant="contained" color="primary">
-                You are unauthorized
-            </ButtonBase>
-            <ButtonBase color="secondary" variant="contained" onClick={startLogin}>
-                Login With Google
-            </ButtonBase>
-        </BoxBase>
+        <SnackbarProvider>
```
Comment (binh1298) [inline src/pages/Login/index.tsx:99]: SnackbarProvider should be in the App.tsx

File: src/pages/Login/index.tsx [hunk 5]
```diff
@@ -73,14 +96,107 @@ const Login = () => {
...
+                    container
+                    item
+                    xs={12}
+                    sm={6}
+                    alignItems="center"
+                    direction="column"
+                    justify="space-between"
+                    style={{ padding: 10 }}
+                >
+                    <BoxBase />
```
Comment (binh1298) [inline src/pages/Login/index.tsx:131]: What is this?

File: src/pages/Login/index.tsx [hunk 6]
```diff
@@ -73,14 +96,107 @@ const Login = () => {
...
+                    <div>
+                        <div
+                            style={{
+                                display: "flex",
+                                flexDirection: "column",
+                                alignItems: "center",
+                                minWidth: 300,
+                                maxWidth: 400,
+                            }}
+                        >
```
Comment (binh1298) [inline src/pages/Login/index.tsx:141]: Please use BoxBase

File: src/pages/Login/index.tsx [hunk 7]
```diff
@@ -73,14 +96,107 @@ const Login = () => {
...
+                            <TypographyBase variant="h3" color="primary">
+                                Sign in
+                            </TypographyBase>
+                            <div style={{ height: 20 }} />
```
Comment (binh1298) [inline src/pages/Login/index.tsx:145]: Please use BoxBase

File: src/pages/Login/index.tsx [hunk 8]
```diff
@@ -73,14 +96,107 @@ const Login = () => {
...
+                                alignItems="center"
+                                justifyContent="center"
+                            >
+                                <hr
+                                    style={{
+                                        border: "solid 1px",
+                                        margin: "auto 20px",
+                                        width: "70px",
+                                    }}
+                                />
```
Comment (binh1298) [inline src/pages/Login/index.tsx:175]: Please make this into a component and a story

File: src/pages/Login/index.tsx [hunk 9]
```diff
@@ -73,14 +96,107 @@ const Login = () => {
...
+                                <p>or</p>
```
Comment (binh1298) [inline src/pages/Login/index.tsx:176]: Please use Typography here

File: src/pages/Login/index.tsx (hunk 4 above)
Comment (binhtt13122000) [reply src/pages/Login/index.tsx:99]: image
I think @thienduc12311  can use useSnackbar to implement easily.
image

File: src/components/Textfields/TextFieldBase/TextfieldBase.stories.tsx [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import TextfieldBase, { TexfieldBaseProps } from ".";
+import TextfieldBase, { TextfieldBaseProps } from ".";
```
Comment (binhtt13122000) [inline src/components/Textfields/TextFieldBase/TextfieldBase.stories.tsx:1]: Thanks for help me to fix this =)))

Comment (kien123456k) [reply src/components/Textfields/TextFieldBase/TextfieldBase.stories.tsx:1]: you are busted :))

File: src/components/Textfields/NormalTextField/index.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
+import { InputAdornment } from "@material-ui/core";
+import TextFieldBase, { TextfieldBaseProps } from "src/components/Textfields/TextFieldBase";
+
+export interface NormalTextFieldProps extends TextfieldBaseProps {
+    icon?: React.ReactNode;
+}
+
+const NormalTextField = (props: NormalTextFieldProps) => {
```
Comment (binh1298) [inline src/components/Textfields/NormalTextField/index.tsx:8]: I think we should not name it "normal" 
Please think of another name using this component's feature/purpose

File: src/pages/Login/index.tsx [hunk 12]
```diff
@@ -73,14 +94,77 @@ const Login = () => {
...
+                            display="flex"
+                        >
+                            <img src={fcodeImage} width="50%" />
+                        </BoxBase>
+                    </Grid>
+                </Hidden>
+
+                <Grid
+                    container
+                    item
```
Comment (binh1298) [inline src/pages/Login/index.tsx:116]: I think you should check this again

File: src/pages/Login/index.tsx [hunk 13]
```diff
@@ -1,9 +1,22 @@
...
+import GoogleButton from "src/components/Buttons/GoogleButton";
+import useSnackbar from "src/components/SnackBars/useSnackbar";
+import NormalTextField from "src/components/Textfields/NormalTextField";
+import ThematicBreak from "src/components/ThematicBreak";
+import TypographyBase from "src/components/Typography/TypographyBase";
+
+import fcodeImage from "src/assets/fcode.png";
+import LocalStorageUtils from "src/utils/LocalStorageUtils";
+
+// import LocalStorageUtils from "src/utils/LocalStorageUtils";
```
Comment (binh1298) [inline src/pages/Login/index.tsx:19]:
```suggestion
```

File: src/components/Textfields/IconTextField/index.tsx [hunk 14]
```diff
@@ -0,0 +1,22 @@
+import { InputAdornment } from "@material-ui/core";
+import TextFieldBase, { TextfieldBaseProps } from "src/components/Textfields/TextFieldBase";
+
+export interface IconTextFieldProps extends TextfieldBaseProps {
+    icon?: React.ReactNode;
+}
+
+const IconTextField = (props: IconTextFieldProps) => {
```
Comment (binh1298) [inline src/components/Textfields/IconTextField/index.tsx:8]: I'm not sure if this should be made into a component, this should be a story of textfieldbase?

File: src/pages/Login/index.tsx [hunk 15]
```diff
@@ -73,14 +92,76 @@ const Login = () => {
...
+                </Hidden>
+
+                <Grid
+                    container
+                    xs={12}
+                    sm={6}
+                    alignItems="center"
+                    direction="column"
+                    justify="space-between"
+                    style={{ padding: 10 }}
```
Comment (binh1298) [inline src/pages/Login/index.tsx:119]: I don't think you understand how to use Grid, please read the docs carefully.

File: src/pages/Login/index.tsx (hunk 5 above)
Comment (binh1298) [reply src/pages/Login/index.tsx:120]: I don't understand the purpose of this @@

**Review by binh1298 — APPROVED** [review]

**Pull request merged**